pkg/internal/infrastructure/terraform.go

```diff
@@ -440,6 +462,43 @@ func findDomainCounts(cluster *controller.Cluster, infra *extensionsv1alpha1.Inf
 	}, nil
 }
 
+// isPrimaryAvailabilitySetRequired determines if a cluster primary AvailabilitySet is required.
+func isPrimaryAvailabilitySetRequired(infra *extensionsv1alpha1.Infrastructure, config *api.InfrastructureConfig, cluster *controller.Cluster) (bool, error) {
+	if config.Zoned {
+		return false, nil
+	}
+	if cluster.Shoot == nil {
+		return false, errors.New("Can't determine if primary availability set is required as cluster.Shoot is not set")
+	}
+
+	hasVmoAnnotation := helper.HasShootVmoAlphaAnnotation(cluster.Shoot.Annotations)
+
+	// If the infrastructureStatus is not exists that mean it is a new Infrastucture.
+	if infra.Status.ProviderStatus == nil {
+		if hasVmoAnnotation {
+			return false, nil
+		}
+		return true, nil
+	}
+
+	// If the infrastructureStatus already exists that mean the Infrastucture is already created.
+	infrastructureStatus, err := helper.InfrastructureStatusFromInfrastructure(infra)
+	if err != nil {
+		return false, err
+	}
+
+	if len(infrastructureStatus.AvailabilitySets) > 0 {
+		if _, err := helper.FindAvailabilitySetByPurpose(infrastructureStatus.AvailabilitySets, api.PurposeNodes); err == nil {
+			if hasVmoAnnotation {
+				return false, errors.New("Can't use vmss orcestration mode VM as this cluster already used an availability set")
```

Comment:
+ `orchestration` typo
+ small case for the error (`cannot`)

Reply:
done